tests/lib/nested.sh

```diff
@@ -33,15 +64,15 @@ nested_wait_for_ssh() {
     done
 }
 
-nested_wait_for_no_ssh() {
-    local retry wait
-    retry=200
-    wait=1
+nested_retry_until_success() {
+    local retry="$1"
+    local wait="$2"
+    shift 2
 
-    while nested_exec true; do
+    while ! nested_exec "$@"; do
```

Comment:
why not just `until` directly? 

```suggestion
    until nested_exec "$@"; do
```